syndesis/plugins/org.fusesource.ide.syndesis.extension.ui/src/org/fusesource/ide/syndesis/extensions/ui/wizards/SyndesisExtensionProjectCreatorRunnable.java

```diff
@@ -344,32 +202,14 @@ private void openSyndesisConfiguration(IProject project, IProgressMonitor monito
 	private void openCamelContextFile(IProject project, IProgressMonitor monitor) {
 		if (project != null) {
 			final IFile[] holder = new IFile[1];
-			searchCamelContextXMLFile(project, holder);
+			
+			// first looks for xml camel file
+			BasicProjectCreatorRunnableUtils.searchCamelContextXMLFile(project, holder);
```

Comment:
Based on your refactor, it seems that we don't use anymore an array holder if the method is returning directly the value

Reply:
I did not grok why we used a holder here as we always seem to return only one file. That's why I did not remove it.

Reply:
I think that it was to avoid usage of an attribute in the class to pass values from inside an org.eclipse.swt.widgets.Display.asyncExec(Runnable) call to outside of it.